### PR TITLE
Fix #3842 Entity class permissions not deleted when an entity metadata is deleted

### LINKED ISSUE #3842: Entity class permissions not deleted when an entity metadata is deleted
- Click delete entitymetadata + data in the dataexplorer
- Goto Admin > Permisison Manager > Entity Class Permissions

Expected: The Permissions for the deleted entity are gone.
Observed: Permissions stil exist.

If you then upload an entity with the same name it automatically gets the permissions set for the deleted entity

### PR DESCRIPTION
**Pull request by erwinwinder**

### CODE REVIEW

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java [hunk 1]
```diff
@@ -138,17 +140,38 @@ public void deleteEntityMeta(String entityName)
...
 	}
 
+	private void deleteEntityPermissions(String entityName)
+	{
+		List<String> authorities = SecurityUtils.getEntityAuthorities(entityName);
+
+		// User permissions
+		if (dataService.hasRepository("UserAuthority"))
+		{
+			Iterable<Entity> userPermissions = dataService.query("UserAuthority").in("role", authorities).findAll();
```
Comment (Mark-de-Haan) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java]: do we want to use UserAuthority.ENTITYNAME and UserAuthority.ROLE?

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java [hunk 2]
```diff
@@ -138,17 +140,38 @@ public void deleteEntityMeta(String entityName)
...
+			dataService.delete("UserAuthority", userPermissions);
+		}
+
+		// Group permissions
+		if (dataService.hasRepository("GroupAuthority"))
+		{
+			Iterable<Entity> groupPermissions = dataService.query("GroupAuthority").in("role", authorities).findAll();
```
Comment (Mark-de-Haan) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java]: do we want to use GroupAuthority.ENTITYNAME and GroupAuthority.ROLE?

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java (hunk 1 above)
Comment (erwinwinder) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java]: In principle yes.
But here we get dependency problems.
That is also the reason that the method is not added to the PermissionManagerService

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java (hunk 2 above)
Comment (erwinwinder) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java]: In principle yes.
But here we get dependency problems.
That is also the reason that the method is not added to the PermissionManagerService

**Comment by Mark-de-Haan** [comment]
Merge with master please

**Pull request merged**